File: IKeep.Web/ClientApp/src/models/Role.js
import Entity from './Entity';

class Role extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.Name = json.name;
            this.UsersIds = json.usersIds;
        }
        else
        {
            this.Name = "";
            this.UsersIds = [];
        }
    }
}

export default Role;

File: IKeep.Lib/Models/Role.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class Role : Entity
    {
        public string Name { get; set; }
        [JsonIgnore]
        public virtual ICollection<InstallationUser> InstallationUsers { get; set; }
        public List<Guid> InstallationUsersIds
        {
            get
            {
                return InstallationUsers == null ? new List<Guid>() : InstallationUsers.Select(x => x.Id).ToList();
            }
        }
    }
}
